pkg/image/admission/admission_test.go

```diff
@@ -0,0 +1,298 @@
+package admission
+
+import (
+	"fmt"
+	"testing"
+
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	imagetest "github.com/openshift/origin/pkg/quota/image/testutil"
+
+	kadmission "k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	clientsetfake "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/fake"
+)
+
+func TestAdmitImageStreamMapping(t *testing.T) {
+	tests := map[string]struct {
+		imageStreamMapping *imageapi.ImageStreamMapping
+		limitRange         *kapi.LimitRange
+		shouldAdmit        bool
+		operation          kadmission.Operation
+	}{
+		"new ism, no limit range": {
+			imageStreamMapping: getImageStreamMapping(),
+			operation:          kadmission.Create,
+			shouldAdmit:        true,
+		},
+		"new ism, under limit range": {
+			imageStreamMapping: getImageStreamMapping(),
+			limitRange:         getLimitRange("1Ki"),
+			operation:          kadmission.Create,
+			shouldAdmit:        true,
+		},
+		"new ism, over limit range": {
+			imageStreamMapping: getImageStreamMapping(),
+			limitRange:         getLimitRange("0Ki"),
+			operation:          kadmission.Create,
+			shouldAdmit:        false,
+		},
+	}
+
+	for k, v := range tests {
+		var fakeKubeClient clientset.Interface
+		if v.limitRange != nil {
+			fakeKubeClient = clientsetfake.NewSimpleClientset(v.limitRange)
+		} else {
+			fakeKubeClient = clientsetfake.NewSimpleClientset()
+		}
+
+		plugin, err := NewImageLimitRangerPlugin(fakeKubeClient, nil)
+		if err != nil {
+			t.Errorf("%s failed creating plugin %v", k, err)
+			continue
+		}
+
+		attrs := kadmission.NewAttributesRecord(v.imageStreamMapping,
+			unversioned.GroupKind{},
+			v.imageStreamMapping.Namespace,
+			v.imageStreamMapping.Name,
+			imageapi.Resource("imagestreammappings"),
+			"",
+			v.operation,
+			nil)
+
+		err = plugin.Admit(attrs)
+		if v.shouldAdmit && err != nil {
+			t.Errorf("%s expected to be admitted but recieved errors %#v", k, err)
```

Comment:
typo: recieved

Also would you expect more than 1 error coming back or is %v ok to use?